### ISSUE TITLE
[core] DFA hits a hard limit of 6 iterations without results and logs

### ISSUE THREAD

**Issue by b-3-n** [pmd/pmd#873]
**Affects PMD Version:** 
6.0.0 and greater

**Rule:**

`java-errorprone/dataflowanomalyanalysis`

**Code Sample demonstrating the issue:**
If you take this file: https://github.com/eclipse/che/blob/b39a70f9c8351cdeb947bc83ff20c3989991659d/ide/commons-gwt/src/main/java/org/eclipse/che/ide/util/input/SignalKeyLogic.java and run pmd on it as follows:
```
./run.sh pmd -d ./SignalKeyLogic.java -f xml -R category/java/errorprone.xml
```
You get this output: 

```
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.cache.NoopAnalysisCache <init>
WARNING: This analysis could be faster, please consider using Incremental Analysis: https://pmd.github.io/pmd/pmd_userdocs_getting_started.html#incremental-analysis
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
aktStatus is NULL: maximum Iterations exceeded, abort 6
Jan 22, 2018 5:54:16 PM net.sourceforge.pmd.lang.dfa.SequenceChecker run
SEVERE: aktStatus is NULL: maximum Iterations exceeded, abort 6
<?xml version="1.0" encoding="UTF-8"?>
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd"
    version="6.0.0" timestamp="2018-01-22T17:54:16.306">
<file name="/home/benjamin/Downloads/pmd-bin-6.0.0/bin/SignalKeyLogic.java">
<violation beginline="65" endline="65" begincolumn="16" endcolumn="22" rule="BeanMembersShouldSerialize" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic$Result" variable="keyCode" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#beanmembersshouldserialize" priority="3">
Found non-transient, non-static member. Please mark as transient or provide accessors.
</violation>
<violation beginline="68" endline="68" begincolumn="26" endcolumn="29" rule="BeanMembersShouldSerialize" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic$Result" variable="type" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#beanmembersshouldserialize" priority="3">
Found non-transient, non-static member. Please mark as transient or provide accessors.
</violation>
<violation beginline="71" endline="71" begincolumn="31" endcolumn="39" rule="BeanMembersShouldSerialize" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" variable="userAgent" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#beanmembersshouldserialize" priority="3">
Found non-transient, non-static member. Please mark as transient or provide accessors.
</violation>
<violation beginline="73" endline="73" begincolumn="17" endcolumn="38" rule="AvoidFieldNameMatchingMethodName" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" variable="commandIsCtrl" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#avoidfieldnamematchingmethodname" priority="3">
Field commandIsCtrl has the same name as a method
</violation>
<violation beginline="73" endline="73" begincolumn="25" endcolumn="37" rule="BeanMembersShouldSerialize" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" variable="commandIsCtrl" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#beanmembersshouldserialize" priority="3">
Found non-transient, non-static member. Please mark as transient or provide accessors.
</violation>
<violation beginline="76" endline="76" begincolumn="17" endcolumn="46" rule="BeanMembersShouldSerialize" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" variable="commandComboDoesntGiveKeypress" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#beanmembersshouldserialize" priority="3">
Found non-transient, non-static member. Please mark as transient or provide accessors.
</violation>
<violation beginline="104" endline="192" begincolumn="13" endcolumn="44" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="ret" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DD'-anomaly for variable 'ret' (lines '104'-'192').
</violation>
<violation beginline="104" endline="197" begincolumn="11" endcolumn="42" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="ret" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DD'-anomaly for variable 'ret' (lines '104'-'197').
</violation>
<violation beginline="104" endline="199" begincolumn="11" endcolumn="53" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="ret" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DD'-anomaly for variable 'ret' (lines '104'-'199').
</violation>
<violation beginline="104" endline="300" begincolumn="13" endcolumn="22" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="ret" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'ret' (lines '104'-'300').
</violation>
<violation beginline="112" endline="112" begincolumn="21" endcolumn="24" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
<violation beginline="122" endline="122" begincolumn="28" endcolumn="29" rule="AvoidLiteralsInIfCondition" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#avoidliteralsinifcondition" priority="3">
Avoid using Literals in Conditional Statements
</violation>
<violation beginline="127" endline="300" begincolumn="13" endcolumn="47" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="isIME" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'isIME' (lines '127'-'300').
</violation>
<violation beginline="129" endline="300" begincolumn="13" endcolumn="58" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="commandKey" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'commandKey' (lines '129'-'300').
</violation>
<violation beginline="145" endline="145" begincolumn="25" endcolumn="28" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
<violation beginline="152" endline="152" begincolumn="27" endcolumn="30" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
<violation beginline="159" endline="183" begincolumn="11" endcolumn="89" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="isActuallyCtrlInput" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DD'-anomaly for variable 'isActuallyCtrlInput' (lines '159'-'183').
</violation>
<violation beginline="159" endline="300" begincolumn="17" endcolumn="43" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="isActuallyCtrlInput" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'isActuallyCtrlInput' (lines '159'-'300').
</violation>
<violation beginline="163" endline="163" begincolumn="27" endcolumn="30" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
<violation beginline="166" endline="300" begincolumn="13" endcolumn="38" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="type" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'type' (lines '166'-'300').
</violation>
<violation beginline="170" endline="300" begincolumn="11" endcolumn="37" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="type" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'type' (lines '170'-'300').
</violation>
<violation beginline="172" endline="300" begincolumn="11" endcolumn="41" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="type" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'type' (lines '172'-'300').
</violation>
<violation beginline="176" endline="300" begincolumn="11" endcolumn="39" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="type" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'type' (lines '176'-'300').
</violation>
<violation beginline="182" endline="300" begincolumn="11" endcolumn="36" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="type" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'type' (lines '182'-'300').
</violation>
<violation beginline="185" endline="300" begincolumn="11" endcolumn="39" rule="DataflowAnomalyAnalysis" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" variable="type" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#dataflowanomalyanalysis" priority="5">
Found 'DU'-anomaly for variable 'type' (lines '185'-'300').
</violation>
<violation beginline="202" endline="202" begincolumn="25" endcolumn="28" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
<violation beginline="217" endline="217" begincolumn="27" endcolumn="30" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
<violation beginline="221" endline="221" begincolumn="25" endcolumn="28" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
<violation beginline="270" endline="270" begincolumn="27" endcolumn="30" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
<violation beginline="285" endline="285" begincolumn="25" endcolumn="28" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
<violation beginline="297" endline="297" begincolumn="21" endcolumn="24" rule="NullAssignment" ruleset="Error Prone" package="org.eclipse.che.ide.util.input" class="SignalKeyLogic" method="computeKeySignalType" externalInfoUrl="https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#nullassignment" priority="3">
Assigning an Object to null is a code smell.  Consider refactoring.
</violation>
</file>
</pmd>
```

**Running PMD through:** CLI

**Comment by Fernal73** [comment]
I'm receiving a similar error while running PMD.

The offending source code is
https://github.com/Fernal73/LearnJava/blob/master/Cryptography/cryptic/GCD.java

**Comment by jwgmeligmeyling** [comment]
Seeing the same issue. For now I've excluded the analysis from the category:

```xml
    <rule ref="category/java/errorprone.xml">
        <exclude name="DataflowAnomalyAnalysis"/>
    </rule>
```

**Comment by adangel** [comment]
The rule [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis) has been deprecated in favour of [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment), which is available with PMD 6.26.0.

Therefore this issue wont be fixed.

See also #2647.
